src/client.rs

```diff
@@ -114,16 +114,25 @@ impl<'mpesa, Env: ApiEnvironment> Mpesa<Env> {
             .await?;
         if response.status().is_success() {
             let value = response.json::<Value>().await?;
+            let error_message ="Failed to extract token from the response";
             let access_token = value
                 .get("access_token")
-                .ok_or_else(|| MpesaError::AuthenticationError(value.clone()))?;
+                .ok_or_else(|| MpesaError::AuthenticationError(
+                    ApiError::new(value.to_string(),String::new(),error_message.to_string()
+                )))?;
             let access_token = access_token
                 .as_str()
-                .ok_or_else(|| MpesaError::AuthenticationError(value.clone()))?;
+                .ok_or_else(|| MpesaError::AuthenticationError(
+                    ApiError::new(value.to_string(), String::new(),error_message.to_string())
+                ))?;
             return Ok(access_token.to_string());
         }
         let value = response.json::<Value>().await?;
```

Comment:
Since we are interested in the entire error json object, why not deserialize it from `response.json` directly i.e

```rs
let error = response.json::<ApiError>().await?;
```